curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/handle-a-fulfilled-promise-with-then.english.md

```diff
@@ -0,0 +1,78 @@
+---
+id: 5cdafbd72913098997531681
+title: Handle a Fulfilled Promise with then
+challengeType: 1
+---
+
+## Description
+<section id='description'>
+Promises are most useful when you have a process that takes an unknown amount of time in your code, something asynchronous. Often, a server request. When you make a server request it takes some amount of time, and after it completes you usually want to do something with the response from the server. This can be achieved by using the <code>then</code> method. The <code>then</code> method is executed immediately after your promise is fulfilled with <code>resolve</code>. Here’s an example:
+
+```js
+myPromise.then(result => {
+  // do something with the result.
+});
+```
+
+<code>result</code> comes from the argument given to the <code>resolve</code> method.
+</section>
+
+## Instructions
+<section id='instructions'>
+Use the <code>then</code> method on your promise. Log its argument to the console.
+</section>
+
+## Tests
+<section id='tests'>
+
+```yml
+tests:
+  - text: You should call the <code>then</code> method on the promise.
+    testString: assert(code.match(/(makeServerRequest|\))\.then\(\s*(\s*function\s*\(\s*result\s*\)\s*|\s*result\s*=>\s*|\s*\(\s*result\s*\)\s*=>\s*){[^}]*}\s*\)/g));
+  - text: You should log the <code>then</code> argument (<code>result</code>) to the console.
+    testString: assert(code.match(/{[^c]*console\.log\(\s*result\s*\)[^}]*}/g));
```

Comment:
Related to my comment above, Since what the user decides to choose for the parameter name of the callback function (it does not have to be "result"), then why not figure out a way to actually have the test make a call and validate something is logged to the console.  @ValeraS Any suggestions how to do something like this with the tests or another creative way to validate the then method is used to do something specific?

Also, the following would be a completely valid way of using console.log of the result but the regex is not flexible enough to handle it.
```
makeServerRequest.then(result => console.log(result));
```

Reply:
Yes, that would be ideal - I thought about that, but didn't figure anything out and decided to just regex it for now. I think if the regex can accommodate the various syntax it should be fine. We could force the campers to use 'result' as the parameter name or maybe not check the parameter name in the regex. I'll take another look here and see if I can figure something out.

Reply:
You could use `(\w)` instead of `result` in the regex and then obtain the name of the parameter used using `match` in order to reference in a later test.